### PR TITLE
Add a method to retrieve a jira card for failed test

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Retrieve an existing jira issue related to a parent test (aka test suite) and add this issue key as an attribute of the junit before upload.

### Motivation
In the scope of TPM improvement we are working on an automatic issue creation for failed tests. As there is no possibility to retrieve an issue from its summary in workflow automation yet, we must pass the information of an already existing ticket at test upload

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/libs/junit_upload.py [hunk 1]
```diff
@@ -246,3 +250,32 @@ def repack_macos_junit_tar(infile, outfile):
...
         for f in os.listdir(tempd):
             outfp.add(os.path.join(tempd, f), arcname=f)
+
+
+def retrieve_jira_card(xmlfile_path):
+    """
+    Search in jira if a card already exist for the given testsuite
+    """
+    from .pipeline_notifications import send_slack_message
+    from jira import JIRA
```
Comment (chouetz) [inline tasks/libs/junit_upload.py:260]: Pending addition of this module to the datadog-ci-uploader image, see https://github.com/DataDog/datadog-agent-buildimages/pull/438

File: tasks/libs/junit_upload.py [hunk 2]
```diff
@@ -246,3 +250,32 @@ def repack_macos_junit_tar(infile, outfile):
...
+
+    tree = ET.parse(xmlfile_path)
+    auth = ("mail", "token")
```
Comment (chouetz) [inline tasks/libs/junit_upload.py:263]: pending creation of a service account (see https://datadog.freshservice.com/support/tickets/372220)

File: tasks/libs/junit_upload.py [hunk 3]
```diff
@@ -246,3 +250,32 @@ def repack_macos_junit_tar(infile, outfile):
...
+    j = JIRA(basic_auth=auth, server="https://datadoghq.atlassian.net/")
+    test_suites = list(tree.iter("testsuite"))
+    if len(test_suites) > 1:
+        message = f"Found several test suites in a single junit file: {[x.attrib['name'] for x in test_suites]}"
+        send_slack_message("#agent-test-alerts", message)
+    suite_name = test_suites[0].attrib["name"]
+    search_query = f'project = "CI Improvement" and summary ~ "{suite_name}" and status != Done'
+    try:
+        issues = j.search_issues(search_query)
```
Comment (chouetz) [inline tasks/libs/junit_upload.py:272]: I should maybe add a retry here

Comment (chouetz) [reply tasks/libs/junit_upload.py:272]: If we fail finding the card we must not fail the job, so I put a try/except to send a fix value that is managed during the [workflow execution](https://app.datadoghq.com/workflow/42375aaf-9a77-4b93-ad51-9a5f524b570d)

Comment (chouetz) [reply tasks/libs/junit_upload.py:272]: @spencergilbert just checked and the JIRA API already has a [3-retries policy](https://github.com/pycontribs/jira/blob/3.5.2/jira/client.py#L400) with 60s sleep time

File: tasks/libs/junit_upload.py (hunk 1 above)
Comment (chouetz) [reply tasks/libs/junit_upload.py:260]: Included in latest image

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**